Intersect.Editor/Core/Graphics.cs

```diff
@@ -1388,65 +1388,123 @@ bool alternate
                         continue;
                     }
 
-                    if (TextUtils.IsNone(resource.Initial.Graphic))
+                    if (resource.States.Count < 1)
                     {
                         continue;
                     }
 
-                    if (resource.Initial.GraphicFromTileset)
+                    var fullVitalResourceGraphic = resource.States.Values
+                        .OrderByDescending(g => g.MaximumHealth)
+                        .FirstOrDefault();
+
+                    if (fullVitalResourceGraphic is not { } resourceGraphic)
                     {
-                        var res = GameContentManager.GetTexture(
-                            GameContentManager.TextureType.Tileset, resource.Initial.Graphic
-                        );
+                        continue;
+                    }
 
-                        if (res == null)
-                        {
-                            continue;
-                        }
+                    // we have the graphic, now lets build based on type
+                    switch(resourceGraphic.TextureType)
+                    {
+                        case ResourceTextureSource.Resource:
+                            {
+                                var texture = GameContentManager.GetTexture(
+                                    GameContentManager.TextureType.Resource, resourceGraphic.TextureName
+                                );
 
-                        float xpos = x * Options.Instance.Map.TileWidth + xoffset;
-                        float ypos = y * Options.Instance.Map.TileHeight + yoffset;
-                        if ((resource.Initial.Height + 1) * Options.Instance.Map.TileHeight > Options.Instance.Map.TileHeight)
-                        {
-                            ypos -= (resource.Initial.Height + 1) * Options.Instance.Map.TileHeight - Options.Instance.Map.TileHeight;
-                        }
+                                if (texture == null)
+                                {
+                                    continue;
+                                }
 
-                        if ((resource.Initial.Width + 1) * Options.Instance.Map.TileWidth > Options.Instance.Map.TileWidth)
-                        {
-                            xpos -= ((resource.Initial.Width + 1) * Options.Instance.Map.TileWidth - Options.Instance.Map.TileWidth) / 2;
-                        }
+                                float xpos = x * Options.Instance.Map.TileWidth + xoffset;
+                                float ypos = y * Options.Instance.Map.TileHeight + yoffset;
```

Comment:
This "base" calculation is common between all 3 cases, please extract the common code